### PR TITLE
Small fixes to HTTP Client based on PR #362

### PR DESCRIPTION
**Pull request by sbunciak**

### CODE REVIEW

File: rest/src/main/java/org/jboss/pnc/rest/trigger/BuildTriggerer.java [hunk 1]
```diff
@@ -178,39 +181,22 @@ private void signalBpmEvent(String uri) {
...
         HttpPost request = new HttpPost(uri);
         request.addHeader("Authorization", getAuthHeader());
         log.info("Executing request " + request.getRequestLine());
 
-        try (CloseableHttpResponse response = httpclient.execute(request)) {
+        try (CloseableHttpResponse response = httpClient.execute(request)) {
             log.info(response.getStatusLine());
             response.close();
-            httpclient.close();
+            httpClient.close();
```
Comment (slaskawi) [inline rest/src/main/java/org/jboss/pnc/rest/trigger/BuildTriggerer.java]: I guess we should use try with resources also here right?

File: rest/src/main/java/org/jboss/pnc/rest/trigger/BuildTriggerer.java [hunk 2]
```diff
@@ -178,39 +181,22 @@ private void signalBpmEvent(String uri) {
...
-        try {
-            sslSF = new SSLConnectionSocketFactory(builder.build(), SSLConnectionSocketFactory.ALLOW_ALL_HOSTNAME_VERIFIER);
-        } catch (KeyManagementException | NoSuchAlgorithmException e1) {
-            e1.printStackTrace();
-        }
 
-        CloseableHttpClient httpclient = HttpClients.custom().setSSLSocketFactory(sslSF)
-                .setHostnameVerifier(new AllowAllHostnameVerifier()).build();
+        /*
+         * TODO: Do not ignore certificates, rather setup servers properly.
```
Comment (slaskawi) [inline rest/src/main/java/org/jboss/pnc/rest/trigger/BuildTriggerer.java]: This comment should be removed IMO

File: rest/src/main/java/org/jboss/pnc/rest/trigger/BuildTriggerer.java (hunk 1 above)
Comment (sbunciak) [inline rest/src/main/java/org/jboss/pnc/rest/trigger/BuildTriggerer.java]: @slaskawi you mean something like the following?
```java
    try (CloseableHttpClient httpClient = HttpUtils.getPermissiveHttpClient()) {
        try (CloseableHttpResponse response = httpClient.execute(request)) {
            log.info(response.getStatusLine());
        }
    } catch (IOException e) {
        log.error("Error occured executing the callback.", e);
    }
```

Comment (slaskawi) [inline rest/src/main/java/org/jboss/pnc/rest/trigger/BuildTriggerer.java]: Yeah I think so... BTW it looks pretty awkward that a response needs calling close...

**Pull request merged**